server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/AllocationBalancingRoundMetrics.java

```diff
@@ -0,0 +1,158 @@
+
+    // gauges that measure current utilization
+    public static final String NUMBER_OF_SHARDS_METRIC_NAME = "es.allocator.balancing_round.shard_count";
+    public static final String DISK_USAGE_BYTES_METRIC_NAME = "es.allocator.balancing_round.disk_usage_bytes";
+    public static final String WRITE_LOAD_METRIC_NAME = "es.allocator.balancing_round.write_load";
+    public static final String TOTAL_WEIGHT_METRIC_NAME = "es.allocator.balancing_round.total_weight";
+
+    private final LongCounter balancingRoundCounter;
+    private final LongCounter shardMovesCounter;
+
+    /**
+     * The current view of the last period's summary
+     */
+    private final AtomicReference<Map<String, NodesWeightsChanges>> nodeNameToWeightChangesRef = new AtomicReference<>();
+
+    /**
+     * Whether metrics sending is enabled
+     */
+    private volatile boolean enableSending = false;
+
+    public static final AllocationBalancingRoundMetrics NOOP = new AllocationBalancingRoundMetrics(MeterRegistry.NOOP);
+
+    private final MeterRegistry meterRegistry;
+
+    public AllocationBalancingRoundMetrics(MeterRegistry meterRegistry) {
+        this.meterRegistry = meterRegistry;
+
+        this.balancingRoundCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_BALANCING_ROUNDS_METRIC_NAME,
+            "Current number of balancing rounds",
+            "unit"
+        );
+        this.shardMovesCounter = meterRegistry.registerLongCounter(
+            NUMBER_OF_SHARD_MOVES_METRIC_NAME,
+            "Current number of shard moves",
+            "{shard}"
+        );
+        this.nodeNameToWeightChangesRef.set(Map.of());
+
+        meterRegistry.registerLongsGauge(NUMBER_OF_SHARDS_METRIC_NAME, "Current number of shards", "unit", this::getShardCount);
+        meterRegistry.registerDoublesGauge(DISK_USAGE_BYTES_METRIC_NAME, "Disk usage in bytes", "unit", this::getDiskUsage);
+        meterRegistry.registerDoublesGauge(WRITE_LOAD_METRIC_NAME, "Write load", "1.0", this::getWriteLoad);
+        meterRegistry.registerDoublesGauge(TOTAL_WEIGHT_METRIC_NAME, "Total weight", "1.0", this::getTotalWeight);
+    }
+
+    public void setEnableSending(boolean enableSending) {
+        this.enableSending = enableSending;
+    }
+
+    public void updateBalancingRoundMetrics(BalancingRoundSummary.CombinedBalancingRoundSummary summary) {
+        assert summary != null : "balancing round metrics cannot be null";
+
+        nodeNameToWeightChangesRef.set(summary.nodeNameToWeightChanges());
+        if (enableSending) {
+            balancingRoundCounter.incrementBy(summary.numberOfBalancingRounds());
+            shardMovesCounter.incrementBy(summary.numberOfShardMoves());
+        }
+    }
+
+    public void clearBalancingRoundMetrics() {
+        nodeNameToWeightChangesRef.set(Map.of());
+    }
+
+    private Map<String, Object> getNodeAttributes(String nodeId) {
+        return Map.of("node_id", nodeId);
+    }
```

Comment:
I think metrics by default get a `node_id` and a corresponding `node_name`, for the node they are emitted from. Perhaps we should use something other than node_id here? Or set `node_name` also?